src/experimental/patterns/ec2-app.ts

```diff
@@ -0,0 +1,307 @@
+ * - Your application should include a build number in its filename.
+ *   This value will change across builds, and therefore create a CloudFormation template difference to be deployed.
+ * - Requires the AWS CLI and `cfn-signal` binaries to be available on the instance, and on the `PATH`.
+ *   AMIgo adds these via the `aws-tools` role.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html#cfn-attributes-updatepolicy-rollingupdate
+ * @see https://github.com/guardian/testing-asg-rolling-update
+ *
+ * @experimental
+ */
+export class GuEc2AppExperimental extends GuEc2App {
+  constructor(scope: GuStack, props: GuEc2AppExperimentalProps) {
+    const { minimumInstances, maximumInstances = minimumInstances * 2 } = props.scaling;
+    const { applicationPort } = props;
+    const { region, stackId } = scope;
+
+    super(scope, {
+      ...props,
+      updatePolicy: UpdatePolicy.rollingUpdate({
+        maxBatchSize: maximumInstances,
+        minInstancesInService: minimumInstances,
+        minSuccessPercentage: 100,
+        waitOnResourceSignals: true,
+
+        /*
+        If a scale-in event fires during an `AutoScalingRollingUpdate` operation, the update could fail and rollback.
+        For this reason, we suspend the `AlarmNotification` process, else availability of a service cannot be guaranteed.
+        Consequently, services cannot scale-out during deployments.
+        If AWS ever supports suspending scale-out and scale-in independently, we should allow scale-out.
+         */
+        suspendProcesses: [ScalingProcess.ALARM_NOTIFICATION],
+      }),
+    });
+
+    const { autoScalingGroup, targetGroup } = this;
+    const { userData, role } = autoScalingGroup;
+    const cfnAutoScalingGroup = autoScalingGroup.node.defaultChild as CfnAutoScalingGroup;
+
+    cfnAutoScalingGroup.desiredCapacity = minimumInstances.toString();
+
+    /*
+    Enable all ASG metrics for improved observability.
+    Specifying `granularity` without specifying `metrics` results in all metrics being enabled.
+    See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-metricscollection.html
+     */
+    cfnAutoScalingGroup.metricsCollection = [
+      {
+        granularity: "1Minute",
+      },
+    ];
+
+    // TODO are these sensible values?
+    const signalTimeoutSeconds = Math.max(
+      targetGroup.healthCheck.timeout?.toSeconds() ?? 0,
+      cfnAutoScalingGroup.healthCheckGracePeriod ?? 0,
+      Duration.minutes(5).toSeconds(),
+    );
+
+    const currentRollingUpdate = cfnAutoScalingGroup.cfnOptions.updatePolicy?.autoScalingRollingUpdate;
+
+    cfnAutoScalingGroup.cfnOptions.updatePolicy = {
+      autoScalingRollingUpdate: {
+        ...currentRollingUpdate,
+        pauseTime: Duration.seconds(signalTimeoutSeconds).toIsoString(),
+      },
+    };
+
+    cfnAutoScalingGroup.cfnOptions.creationPolicy = {
```

Comment:
Minor: do we need to add the creation policy as part of this change? I think it could be added as a standalone feature without being coupled to this experimental pattern?

Reply:
Technically it can be its own change. However, this creation policy is configured to keep the ASG resource in `CREATE_IN_PROGRESS` until the correct number of `SUCCESS` signals are received. The signals need to be explicitly sent, which we've starting doing in this branch, so I think coupling to the experimental pattern is simplest. WDYT?

Reply:
Ah yes, sorry, I now see that it's not a useful feature without the `SUCCESS` signals being sent - let's leave it in!